UI/platform-osx.mm

```diff
@@ -62,10 +62,13 @@ string GetDefaultVideoSavePath()
     NSURL *url = [fm URLForDirectory:NSMoviesDirectory inDomain:NSUserDomainMask appropriateForURL:nil create:true
                                error:nil];
 
-    if (!url)
-        return getenv("HOME");
+    if (!url) {
+        NSString *path = [NSString pathWithComponents:@[getenv("HOME"), @"/OBS Studio"]];
+        return path;
```

Comment:
You can't simply return an `NSString` pointer (which is reference counted by the ObjC runtime) as a `std::string` object (which either uses C++ semantics and memory management).

Instead you need to take a `char` pointer reference of the `NSString` and use that to initialise a `std::string` - this is usually done by using the `UTF8String` property of the `NSString` object.

The compiler will insert the required code to ensure this all works (the `char` pointer will become invalid once the `NSString` object is automatically deallocated by the ObjC runtime once the function goes out of scope) so it will implicitly create a `std::string` with a copy of that `char` array before returning the string and then _that_ string will be used in the copy constructor of the target string (so you will have the entire string being copied around in memory at least twice).